api/application.proto

```diff
@@ -28,9 +29,9 @@ option go_package = "go.thethings.network/lorawan-stack/pkg/ttnpb";
 
 // Application is the message that defines an Application in the network.
 message Application {
-  ApplicationIdentifiers ids = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false];
-  google.protobuf.Timestamp created_at = 2 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
-  google.protobuf.Timestamp updated_at = 3 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  ApplicationIdentifiers ids = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false, (validate.rules).message.required = true];
+  google.protobuf.Timestamp created_at = 2 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true, (validate.rules).message.required = true];
```

Comment:
What does this mean? It's non-nullable already, so empty is not valid? These values are generated by the server, right?

Reply:
When GoGo protoc plugin is used to generate Go code and `(gogoproto.nullable) = false` is set, this indeed does not generate anything.
The reasons to have these:
1. Validation generated by plugins not supporting GoGo extensions (e.g. original Go protoc or other langugage plugin)
2. In case we ever decide to make any of the "required" non-nullable fields nullable, we will be able to automatically ensure validity of the entities

Reply:
OK, I get it for the code generation.

But are _these_ fields required? When you create an application, it's fine to have and even better to have `"created_at":nil` in, say, Node.JS, right? Instead of some language default timestamp that doesn't mean anything.